### PR TITLE
Add separate targets for unit and e2e tests

### PR DESCRIPTION
**Pull request by rexagod**
Folks should be able to selectively test out e2e and unit tests. Also,
added coverage under tests (to avoid significantly larger computing
costs by the CI).

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

### CODE REVIEW

**Comment by rexagod** [comment]
Context: https://github.com/openshift/release/pull/27907#discussion_r868199963

File: Makefile [hunk 1]
```diff
@@ -46,12 +48,25 @@ lint:
...
+    source $(ENVTEST_ASSETS_DIR)/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR)
+
+test: generate fmt vet manifests setup-envtest
+	go get golang.org/x/tools/cmd/cover; \
+ 	NOOBAA_CORE_IMAGE={NOOBAA_CORE_IMAGE} NOOBAA_DB_IMAGE={NOOBAA_DB_IMAGE} go test -v -coverprofile coverage.out ./...; \
+	$(MAKE) -f $(MAKEFILE) coverage
+
+unit-test: generate fmt vet manifests
+	go get golang.org/x/tools/cmd/cover; \
+	NOOBAA_CORE_IMAGE={NOOBAA_CORE_IMAGE} NOOBAA_DB_IMAGE={NOOBAA_DB_IMAGE} go test -v -coverprofile coverage.out ./controllers; \
```
Comment (naveenpaul1) [inline Makefile:63]: unit test only for controllers folder? It will block us achieving maximum coverage

**Review by naveenpaul1 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (rexagod) [reply Makefile:63]: We can add the folders in our repository as we go along, while making sure that `tests` is excluded. IIRC right now only `./controllers` has unit tests.

**Review by naveenpaul1 — CHANGES_REQUESTED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (jarrpa) [reply Makefile:63]: You could also just do `$(go list ./... | grep -v tests)` or some equivalent.

**Pull request merged**